### PR TITLE
Update GH Pages setup example to include the latest actions version

### PR DESCRIPTION
**Pull request by esimkowitz**
## What I did

This adds the latest version of the GH Pages deploy action, as the older version was failing. It also adds an optional task to install Yarn Modern, if a repo requires it.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

1. Run a deployment of the provided actions template in GitHub
2. Confirm that the site deploys successfully.

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by esimkowitz** [comment]
@jonniebigodes checking in as I haven't heard any updates in 2 weeks, do you need me to make any changes to help get this checked in? It's a simple doc change

**Comment by jonniebigodes** [comment]
@esimkowitz, sorry for the radio silence; I was otherwise engaged with other documentation issues that required my immediate attention and left this pull request without a review. If you're ok with it, I'll take a look at it next week and follow up with you on any feedback that may be required. Sounds good to you?

Have a great day.

Stay safe

File: docs/snippets/common/ghp-github-action.yml.mdx [hunk 1]
```diff
@@ -27,7 +27,11 @@ jobs:
       # Set up Node
       - uses: actions/setup-node@v4
         with:
-          node-version: '16.x'
+          node-version: '20.x'
+
+      # If using Yarn Modern, you also need to add the following action to install your configured Yarn version:
+      # - name: Install Yarn
+      #   run: corepack enable && yarn install
```
Comment (jonniebigodes) [inline docs/snippets/common/ghp-github-action.yml.mdx:34]: @esimkowitz If you're ok with it, could you move the Corepack step above? Before the node setup action, turning it into something as follows:

```yml
name: "Build and Publish Storybook to GitHub Pages"

on:
  push:
    branches:
      - 'your-branch-name' # Replace with the branch you want to deploy from

permissions:
  contents: read
  pages: write
  id-token: write

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      # Enable the steps below if you're working with Corepack's supported package managers (e.g. Yarn 2+, pnpm)
      #- name: Enable Corepack
      #  run: corepack enable
      - uses: actions/setup-node@v4
        with:
          node-version: 20
      - run: yarn install
      - uses: bitovi/github-actions-storybook-to-github-pages@v1.0.3
        with:
          install_command: yarn install # default: npm ci
          build_command: yarn build-storybook # default: npm run build-storybook
          path: storybook-static # default: dist/storybook
          checkout: false # default: true
```

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@esimkowitz, once again, I'm sorry for the delayed response. I left a small item for you to look into when you can. Also, if you want, you can check the suggestion enabled [here](https://github.com/jonniebigodes/testing-bitovi-yarn-4/actions/runs/9340281756/workflow).

Let me know, and we'll go from there.


Looking forward to hearing from you.

Have a fantastic rest of your weekend.

Stay safe

File: docs/snippets/common/ghp-github-action.yml.mdx (hunk 1 above)
Comment (esimkowitz) [reply docs/snippets/common/ghp-github-action.yml.mdx:34]: I think corepack is bundled with Nodejs, I haven't tested if it's available before NodeJS gets installed

Comment (jonniebigodes) [reply docs/snippets/common/ghp-github-action.yml.mdx:34]: @esimkowitz I've tested it, and corepack is already bundled with Node.js, so we don't need it here.

**Comment by jonniebigodes** [comment]
@esimkowitz, thank you for putting together this pull request and helping us improve the documentation. We appreciate it 🙏 ! However, the changes you're introducing have already been added to a recent pull request. I hope you don't take it the wrong way, and I look forward to your next contribution.

Have a great day.

Stay safe.

**Pull request closed without merging**